src/databricks/labs/ucx/assessment/clusters.py

```diff
@@ -46,6 +47,8 @@ class ClusterInfo:
     creator: str | None = None
     """User-name of the creator of the cluster, if known."""
 
+    __id_attributes__: ClassVar[Sequence[str]] = ("cluster_id",)
```

Comment:
nit: do we really need `ClassVar` annotation in every single class? using a method in interface-on-read/protocol is a more concise option - https://typing.readthedocs.io/en/latest/spec/protocol.html

Reply:
We are already using a protocol: https://github.com/databrickslabs/ucx/blob/2e5869d80811ce8a7449b9dc095b7a9423822ee1/src/databricks/labs/ucx/progress/history.py#L21

There are a few options here:

 - Inherit from the protocol. This allows the annotation to be left out (and implied from the base class). This works, except unfortunately the mock SQL backend has a bug that means our entity classes are not allowed to inherit from anything.
 - Leave out the annotation entirely. This should probably work, but mypy fails to recognise that the (class) field has the expected type for the protocol.
 - Just use the inner annotation. This should possibly work, but mypy fails to recognise that the (class) field has the expected type for the protocol.
 - Drop the `ClassVar` thing on the protocol. Aside from being incorrect (I think?) mypy rejects it.